mautic/mautic
# Fix for scheduled events If contact was already removed from campaign

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | y
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/3929
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
So ugly bug,
Based on my clients issie with scheduled events
and also this issue https://github.com/mautic/mautic/issues/3929
Error log:
```
[  [Doctrine\ORM\EntityNotFoundException]                                                     
  Entity of type 'Mautic\CampaignBundle\Entity\LeadEventLog' for IDs id(xxx) was not found  ](url)
```

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Create campaign
2. Create two scheduled action 
- first - Campaign update - remove from this campaign - 1 min delay
- second - Modify tags - 1 min delay
3. Add contact to campaign
4. Run `php app/console m:c:t`
5. Wait 1 min and run `php app/console m:c:t`
6. Return error because triggerScheduledEvents try execute both events, even if first even already removed contact  from campaign

#### Steps to tests:
1. Repeat all steps to reproduce
2. See If works without errors
